src/Altinn.Notifications/Validators/Rules/RecipientRules.cs

```diff
@@ -0,0 +1,245 @@
+                .Must(mobileNumber => MobileNumberHelper.IsValidMobileNumber(mobileNumber))
+                .When(mobileNumber => !string.IsNullOrEmpty(mobileNumber))
+                .WithMessage("Invalid mobile number format.");
+        });
+    }
+
+    private static IRuleBuilderOptions<T, string?> MustBeValidNationalIdentityNumber<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+        .ChildRules(nationalIdentityNumber =>
+        {
+            nationalIdentityNumber.RuleFor(nin => nin)
+                .Must(nin => nin?.Length == _nationalIdentityNumberLength && nin.All(char.IsDigit))
+                .When(nin => !string.IsNullOrEmpty(nin))
+                .WithMessage($"National identity number must be {_nationalIdentityNumberLength} digits long.");
+        });
+    }
+
+    private static IRuleBuilderOptions<T, string?> MustBeValidOrganizationNumber<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+       .ChildRules(organizationNumber =>
+       {
+           organizationNumber.RuleFor(on => on)
+           .Must(on => on?.Length == _organizationNumberLength && on.All(char.IsDigit))
+            .When(on => !string.IsNullOrEmpty(on))
+            .WithMessage($"Organization number must be {_organizationNumberLength} digits long.");
+       });
+    }
+
+    private static IRuleBuilderOptions<T, RecipientExt> MustNotCombineNationalIdentityNumber<T>(this IRuleBuilder<T, RecipientExt> ruleBuilder)
+    {
+        return ruleBuilder
+        .ChildRules(recipient =>
+        {
+            recipient.RuleFor(r => r)
+            .Must(r => string.IsNullOrEmpty(r.EmailAddress) && string.IsNullOrEmpty(r.MobileNumber) && string.IsNullOrEmpty(r.OrganizationNumber))
+            .When(r => !string.IsNullOrEmpty(r.NationalIdentityNumber))
+            .WithMessage("National identity number cannot be combined with email address, mobile number, or organization number.");
+        });
+    }
+
+    private static IRuleBuilderOptions<T, RecipientExt> MustNotCombineOrganizationNumber<T>(this IRuleBuilder<T, RecipientExt> ruleBuilder)
+    {
+        return ruleBuilder
+        .ChildRules(recipient =>
+        {
+            recipient.RuleFor(r => r)
+                .Must(r => string.IsNullOrEmpty(r.EmailAddress) && string.IsNullOrEmpty(r.MobileNumber) && string.IsNullOrEmpty(r.NationalIdentityNumber))
+                .When(r => !string.IsNullOrEmpty(r.OrganizationNumber))
+                .WithMessage("Organization number cannot be combined with email address, mobile number, or national identity number.");
+        });
+    }
+
+    /// <summary>
+    /// Validated as email address based on the Altinn 2 regex
+    /// </summary>
+    /// <param name="email">The string to validate as an email address</param>
+    /// <returns>A boolean indicating that the email is valid or not</returns>
+    internal static bool IsValidEmail(string? email)
```

Comment:
Doesn't the built-in validator for email do the job?

Reply:
Our regex is a bit more robust than the built-in validator which allowed the following test cases which we expect to fail: user.@example.com, user@exa!mple.com and user@example..com